### PR TITLE
Update README and refactor to support `cargo test`

### PR DESCRIPTION
**Pull request by ranjitjhala**
misc edits to readme

### CODE REVIEW

**Comment by ranjitjhala** [comment]
I have refactored some code to support `cargo test` e.g. see

```
liquid-rust-driver/tests/{pos, neg}.rs
```

File: liquid-rust-typeck/src/lib.rs [hunk 1]
```diff
@@ -115,8 +115,12 @@ impl Checker<'_, '_> {
 
         println!("{:?}", constraint);
         let constraint = constraint.to_fixpoint();
-        println!("{:?}", Fixpoint::check(&constraint));
-        Ok(())
+        let fixpoint_res = Fixpoint::check(&constraint);
+        // println!("FIXPOINT {:?}", fixpoint_res);
```
Comment (nilehmann) [inline liquid-rust-typeck/src/lib.rs:119]:
```suggestion
```

File: liquid-rust-driver/src/callbacks.rs [hunk 2]
```diff
@@ -55,7 +62,8 @@ fn check_crate(tcx: TyCtxt, sess: &Session) -> Result<(), ErrorReported> {
         .iter()
         .map(|(def_id, spec)| {
             if spec.assume {
-                return Ok(());
+                // return Ok(());
```
Comment (nilehmann) [inline liquid-rust-driver/src/callbacks.rs:65]:
```suggestion
```

**Review by nilehmann — CHANGES_REQUESTED** [review]

File: liquid-rust-driver/src/lib.rs [hunk 3]
```diff
@@ -21,9 +21,60 @@ mod resolve;
...
+
+/// Test a single file that should be SAFE
+pub fn test_safe(file: &str) -> bool {
+    test_file(file, liquid_rust_fixpoint::Safeness::Safe)
+}
+
+/// Test a single file that should be SAFE
+pub fn test_unsafe(file: &str) -> bool {
+    test_file(file, liquid_rust_fixpoint::Safeness::Unsafe)
+}
```
[the remark this answers is not in the document]
Comment (ranjitjhala) [reply liquid-rust-driver/src/lib.rs:78]: This is very confusing. 

1. If I put the functions in `liquid-rust-driver/tests/common.rs` then I can do `mod common;` and `common::test_safe(...)` BUT it complains in the definition of `common.rs` that `test_safe` is unused? (Its a warning, the tests actually run...)

2. If I put it in `tests/mod.rs` I simply cannot use them?

Any hint?

**Review by nilehmann — APPROVED** [review]

**Pull request merged**